.github/workflows/ci.yml

```diff
@@ -0,0 +1,21 @@
+name: 'ci'
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  # Linting job - go.
+  go_lint:
```

Comment:
Should we hook up `go_test` while we're here?

Reply:
Done.